sycl/source/detail/queue_impl.hpp

```diff
@@ -378,9 +373,9 @@ class queue_impl {
   /// \param Loc is the code location of the submit call (default argument)
   /// \return a SYCL event representing submitted command group.
   event submit_impl(const function_class<void(handler &)> &CGF,
-                    shared_ptr_class<queue_impl> Self,
+                    const shared_ptr_class<queue_impl> &Self,
                     const detail::code_location &Loc) {
-    handler Handler(std::move(Self), MHostQueue);
+    handler Handler(Self, MHostQueue);
```

Comment:
I believe `handler` should now accept queue by reference.

Reply:
Yes, it should.